signer/tests/devenv/devenv.rs

```diff
@@ -0,0 +1,739 @@
+        .with_stacks_client(stacks_client.clone())
+        .with_mocked_emily_client()
+        .build();
+
+    let (rpc, faucet) = regtest::initialize_blockchain_devenv();
+
+    // No need for deposits here
+    ctx.with_emily_client(|client| {
+        client
+            .expect_get_deposits()
+            .returning(move || Box::pin(std::future::ready(Ok(vec![]))));
+
+        client
+            .expect_get_limits()
+            .times(1..)
+            .returning(|| Box::pin(async { Ok(SbtcLimits::unlimited()) }));
+    })
+    .await;
+
+    let block_observer = BlockObserver {
+        context: ctx.clone(),
+        bitcoin_blocks: testing::btc::new_zmq_block_hash_stream(BITCOIN_CORE_ZMQ_ENDPOINT).await,
+    };
+
+    // We need to wait for the block observer to be up
+    let start_flag = Arc::new(AtomicBool::new(false));
+    let flag = start_flag.clone();
+
+    let mut signal_rx = ctx.get_signal_receiver();
+    let block_observer_handle = tokio::spawn(async move {
+        flag.store(true, Ordering::Relaxed);
+        block_observer.run().await
+    });
+
+    while !start_flag.load(Ordering::SeqCst) {
+        tokio::time::sleep(Duration::from_millis(10)).await;
+    }
+    // At this point, the block observer is ready
+
+    // First, let's check everything works
+    faucet.send_to(1001, &faucet.address);
+    let block_1a = faucet.generate_block();
+
+    // Let's wait for the block observer signal
+    let signal = signal_rx.recv();
+    let Ok(SignerSignal::Event(SignerEvent::BitcoinBlockObserved)) = signal.await else {
+        panic!("Not the right signal")
+    };
+
+    let (bitcoin_tip_original, _) = db.get_chain_tips().await;
+    assert_eq!(block_1a, bitcoin_tip_original.block_hash.into());
+
+    // Now we fork by invalidating the tip and creating a 2 blocks branch
+    rpc.invalidate_block(&block_1a).unwrap();
+    // Need also to add some other tx to ensure we get a different block hash
+    faucet.send_to(1002, &faucet.address);
+    let block_1b = faucet.generate_block();
+    assert_ne!(block_1a, block_1b);
+
+    let block_2b = faucet.generate_block();
+
+    // Let's wait for the block observer signal
+    let signal = signal_rx.recv();
+    let Ok(SignerSignal::Event(SignerEvent::BitcoinBlockObserved)) = signal.await else {
+        panic!("Not the right signal")
+    };
+
+    let (bitcoin_tip_fork, _) = db.get_chain_tips().await;
+    assert_eq!(block_2b, bitcoin_tip_fork.block_hash.into());
+    assert_eq!(
+        bitcoin_tip_original.block_height + 1,
+        bitcoin_tip_fork.block_height
+    );
+
+    let original_in_fork = db
+        .in_canonical_bitcoin_blockchain(&bitcoin_tip_fork, &bitcoin_tip_original)
+        .await
+        .unwrap();
+    assert!(!original_in_fork);
+
+    block_observer_handle.abort();
+    testing::storage::drop_db(db).await;
+}
+
+/// Test checking `getrawtransaction` behaviour after a fork
+#[test_log::test(tokio::test)]
+async fn getrawtransaction_simple_fork() {
```

Comment:
Can't these `getrawtransaction` tests run in CI? If so, we can move them as an integration test in the `sbtc` crate where we have some tests that are actually just bitcoin core behavior tests. We could also move them to the signer crate.